src/Core/Algorithm/Subdivision/LoopSubdivider.cpp

```diff
@@ -0,0 +1,333 @@
+    mesh.set_halfedge_handle( fh_old, heh4 );
+
+    // Re-Set Handles around new Face
+    mesh.set_next_halfedge_handle( heh1, heh3 );
+    mesh.set_next_halfedge_handle( heh3, heh2 );
+
+    mesh.set_face_handle( heh1, fh_new );
+    mesh.set_face_handle( heh2, fh_new );
+    mesh.set_face_handle( heh3, fh_new );
+
+    mesh.set_halfedge_handle( fh_new, heh1 );
+
+    // deal with custom properties
+    copyProps( m_floatProps, heh1, heh4, mesh );
+    copyProps( m_floatProps, heh5, heh3, mesh );
+    copyProps( m_vec2Props, heh1, heh4, mesh );
+    copyProps( m_vec2Props, heh5, heh3, mesh );
+    copyProps( m_vec3Props, heh1, heh4, mesh );
+    copyProps( m_vec3Props, heh5, heh3, mesh );
+    copyProps( m_vec4Props, heh1, heh4, mesh );
+    copyProps( m_vec4Props, heh5, heh3, mesh );
+}
+
+void LoopSubdivider::split_edge( TopologicalMesh& mesh, const TopologicalMesh::EdgeHandle& eh ) {
+    using HeHandle = TopologicalMesh::HalfedgeHandle;
+    using VHandle = TopologicalMesh::VertexHandle;
+    using FHandle = TopologicalMesh::FaceHandle;
+
+    // prepare data
+    HeHandle heh = mesh.halfedge_handle( eh, 0 );
+    HeHandle opp_heh = mesh.halfedge_handle( eh, 1 );
+    HeHandle new_heh;
+    HeHandle opp_new_heh;
+    HeHandle t_heh;
+    VHandle vh;
+    VHandle vh1( mesh.to_vertex_handle( heh ) );
+
+    // new vertex
+    vh = mesh.new_vertex( mesh.property( m_epPos, eh ) );
+
+    // memorize position, since it will be set later
+    mesh.property( m_vpPos, vh ) = mesh.property( m_epPos, eh );
+
+    // Re-link mesh entities
+    if ( mesh.is_boundary( eh ) )
+    {
+        for ( t_heh = heh; mesh.next_halfedge_handle( t_heh ) != opp_heh;
+              t_heh = mesh.opposite_halfedge_handle( mesh.next_halfedge_handle( t_heh ) ) )
+            ;
+    } else
+    {
+        for ( t_heh = mesh.next_halfedge_handle( opp_heh );
+              mesh.next_halfedge_handle( t_heh ) != opp_heh;
+              t_heh = mesh.next_halfedge_handle( t_heh ) )
+            ;
+    }
+
+    new_heh = mesh.new_edge( vh, vh1 );
+    opp_new_heh = mesh.opposite_halfedge_handle( new_heh );
+    mesh.set_vertex_handle( heh, vh );
+
+    mesh.set_next_halfedge_handle( t_heh, opp_new_heh );
+    mesh.set_next_halfedge_handle( new_heh, mesh.next_halfedge_handle( heh ) );
+    mesh.set_next_halfedge_handle( heh, new_heh );
+    mesh.set_next_halfedge_handle( opp_new_heh, opp_heh );
+
+    if ( mesh.face_handle( opp_heh ).is_valid() )
+    {
+        mesh.set_face_handle( opp_new_heh, mesh.face_handle( opp_heh ) );
+        mesh.set_halfedge_handle( mesh.face_handle( opp_new_heh ), opp_new_heh );
+
+        // deal with custom properties
+        interpolateProps( m_floatProps, t_heh, opp_heh, opp_new_heh, 0.5, mesh );
+        interpolateProps( m_vec2Props, t_heh, opp_heh, opp_new_heh, 0.5, mesh );
+        interpolateProps( m_vec3Props, t_heh, opp_heh, opp_new_heh, 0.5, mesh );
+        interpolateProps( m_vec4Props, t_heh, opp_heh, opp_new_heh, 0.5, mesh );
```

Comment:
Well, here we can imagine something better than compute the average of props, what about if there is an hard edge ? I think as a first approx it's ok, but need to test on a mesh with discontinuities on props.

Reply:
Sure, we can imagine different ways of handling attributes. But this will require much more time that what we can spend on this for the moment. The test I made were on closed, smooth object with continuous texture coordinates. As we must go forward with all the ongoing PR, subdividing only smooth attribute is a good starting point. We we'll have a clearly define use case for discontinuous attribs, we will put effort on this.

Reply:
Actually it already delas with hard edges with discontinuities on the props on each side: each side of the edge is interpolated independantly.
If you mean discontinuity along the edge, in the same face, then yes, we would need something else than just linear interpolation. And for the moment, only the ìn_normals` atribute is renormalized.

Reply:
Ok, I was talking about side of the edge, seems ok (next thing is to use a interpolation weight depending on new vertex position)